### PR TITLE
Add unit test for null and exception handling.

### PR DESCRIPTION
**Pull request by WenceyWang**
Test for https://github.com/bcuff/AutoLazy/issues/7 .
I am really in need of this project but I am so poor in writing in IL, so the only way to contribute is to add some test.

### CODE REVIEW

File: src/AutoLazy.Tests/GenericTypeTests.cs [hunk 1]
```diff
@@ -24,13 +29,81 @@ public void MockGeneric_with_int_should_be_lazy()
...
+            {
+                second = target.GetValue();
+            }
+            catch (ExpectedException e)
+            {
+                secondException = e;
+            }
+
+            Assert.IsNotNull(secondException);
+            Assert.AreSame(firstException, secondException );
```
Comment (bcuff) [inline src/AutoLazy.Tests/GenericTypeTests.cs]: `[AutoLazy]` will not save an exception if one is encountered so I would expect this assertion to fail.
